src/main/java/uk/gov/companieshouse/api/strikeoffobjections/controller/ObjectionController.java

```diff
@@ -145,8 +146,11 @@ public ObjectionController(PluggableResponseEntityFactory responseEntityFactory,
         );
 
         try {
-            String objectionId = objectionService.createObjection(requestId, companyNumber, ericUserId, ericUserDetails);
-            ObjectionResponseDTO response = new ObjectionResponseDTO(objectionId);
+            ObjectionResponseDTO response = objectionService.createObjection(requestId, companyNumber, ericUserId, ericUserDetails);
+            ObjectionStatus obStat = response.getStatus();
+            if (obStat != null && obStat.isEligibilityError()) {
```

Comment:
Can this really ever be null at this point in a legitimate scenario? Not sure it can be and if that's the case and it is somehow null, this indicates that something unexpected has happened - we would probably want the code to fail 'noisily', so blowing up with a NullPointerException would be preferable and perfectly fine. Basically, what I'm saying is, I don't think that the check for 'obStat != null' is required.